core_types/src/time.rs

```diff
@@ -1,12 +1,12 @@
-//! The Iso8601 type is defined here. It is used in particular
-//! within ChainHeader to enforce that their timestamps
-//! are defined in a useful and consistent way.
+//! The Iso8601 type is defined here. It is used in particular within ChainHeader to enforce that
```

Comment:
This is a superb file @pjkundert .  I think really we should be abstracting this out as its own rust crate for contribution to the community, and then using it.  That would also increase surface area for removing any bugs in it, if others are using it for orthogonal purposes.